include/fmt/format.h

```diff
@@ -1143,12 +1138,14 @@ inline format_decimal_result<Iterator> format_decimal(Iterator out, UInt value,
 }
 
 template <unsigned BASE_BITS, typename Char, typename UInt>
-inline Char* format_uint(Char* buffer, UInt value, int num_digits,
-                         bool upper = false) {
+FMT_CONSTEXPR20 Char* format_uint(Char* buffer, UInt value, int num_digits,
+                                  bool upper = false) {
   buffer += num_digits;
   Char* end = buffer;
   do {
-    const char* digits = upper ? "0123456789ABCDEF" : data::hex_digits;
+    const char* digits = upper ? "0123456789ABCDEF"
+                               : (is_constant_evaluated() ? "0123456789abcdef"
+                                                          : data::hex_digits);
```

Comment:
Can we make data::hex_digits constexpr instead?

Reply:
But I [tried](https://github.com/fmtlib/fmt/pull/2019#discussion_r527027972) to make them available both at compile-time and at runtime via macro definitions. Right now they are embedding into the static library, so I don't even touch them in this PR to not break the ABI. Maybe such variables should be declared as `constexpr` and be copied to that static data structure for runtime uses and still be chosen for compile-time uses.

Reply:
Also, we can go the same way as with `digits` - by implementing cast specially for compile-time. It's going to be more complex, but it's possible.

Reply:
I mean defining it as

```c++
static constexpr char hex_digits[] = "0123456789abcdef";
```
in the `basic_data` struct. This is a tiny piece of data so we don't need to worry about having it in the header.